sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp

```diff
@@ -55,6 +55,8 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
       {
         throw std::invalid_argument("Name cannot be empty");
       }
+
+      Properties.Name = Name;
```

Comment:
Can you call Properties ctor instead of this?
like
```cpp
KeyVaultSecret(std::string name, std::string value)
        : Name(std::move(name)), Value(std::move(value)), Properties(Name)
    {}
```